pkg/connection/config_test.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connection
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func cleanConfigDir(t *testing.T) func() {
+	t.Helper()
+	tmpDir := t.TempDir()
+	origConfigPath := ConfigPath
+	ConfigPath = tmpDir
+	return func() {
+		ConfigPath = origConfigPath
+		os.RemoveAll(tmpDir)
+	}
+}
+
+func TestActiveSettings(t *testing.T) {
+	defer cleanConfigDir(t)()
```

Comment:
Usually in tests `t.Cleanup` is preferred to `defer` because it works even if the test becomes nested inside another. I'm not sure about the behavior of `defer` when `t.Fatal` is called, but it definitely works as expected with `t.Cleanup`.

Reply:
ok